craft_providers/bases/buildd.py

```diff
@@ -496,18 +496,32 @@ def _disable_and_wait_for_snap_refresh(
 
         _check_deadline(deadline)
         # TODO: run `snap refresh --hold` once during setup (`--hold` is not yet stable)
-        executor.execute_run(
-            ["snap", "set", "system", f"refresh.hold={hold_time.isoformat()}Z"],
-            capture_output=True,
-            check=True,
-        )
+        try:
+            executor.execute_run(
+                ["snap", "set", "system", f"refresh.hold={hold_time.isoformat()}Z"],
+                capture_output=True,
+                check=True,
+            )
+        except subprocess.CalledProcessError as error:
```

Comment:
A bit late to the party (I see that this is already merged and that's fine) but maybe we should generalize this error handling a bit? Ideally we would always want to see the output of a failed command, right? Seems a bit annoying to have to remember to add try/except guards around every call to `execute_run()`
One idea would be to wrap the whole `setup()` call on a stop-gap try/catch